modules/apps/commerce/commerce-frontend-js/test/quantitySelector/QuantitySelector.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @liferay/liferay/imports-first */
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+jest.mock(
+	'../../src/main/resources/META-INF/resources/components/quantity_selector/utils/index'
+);
+
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render} from '@testing-library/react';
+import React from 'react';
+
+import QuantitySelector from '../../src/main/resources/META-INF/resources/components/quantity_selector/QuantitySelector';
+import * as Utils from '../../src/main/resources/META-INF/resources/components/quantity_selector/utils/index';
+
+describe('Quantity Selector', () => {
+	describe('When QuantitySelector is loaded', () => {
+		beforeEach(() => {
+			jest.resetAllMocks();
+		});
+
+		afterEach(() => {
+			cleanup();
+		});
+
+		it('if allowedQuantity is different from default [] the options should display the the correct option', () => {
```

Comment:
The idea of [Behavior-Driven Development](https://en.wikipedia.org/wiki/Behavior-driven_development) and DSLs based on `describe` and `it` is that they should produce valid English sentences that sound grammatical when read out loud. So:

- "describe Quantity Selector": good.
- "describe When Quantity Selector is loaded": good (although I'd dump the capital "W" in "When" because it is not a proper noun nor the first word of the sentence).
- "it if allowedQuantity is different from default [] the options should display the the correct option": bad (also note the repeated "the" in there).

In general if you have a "should" in there that is a cue to reword anyway, because you can usually assert it without it. We even have [a lint rule](https://github.com/liferay/liferay-frontend-projects/blob/master/projects/eslint-config/plugins/liferay/docs/rules/no-it-should.md) that bans "should" as the first word in an `it` description.

So for these I'd suggest trying to reword them to start with a verb:

```js
it('displays the correct option when allowedQuantity is different from the default', () => {
```

And similar for the others in this file.